pkg/ui/src/views/shared/components/summaryBar/index.tsx

```diff
@@ -156,20 +159,61 @@ export function SummaryStatBreakdown(props: SummaryStatBreakdownProps & {childre
  * SummaryMetricStat is a helpful component that creates a SummaryStat where
  * metric data is automatically derived from a metric component.
  */
-export function SummaryMetricStat(propsWithID: SummaryStatProps & { id: string } & { children?: React.ReactNode }) {
+export function SummaryMetricStat(propsWithID: SummaryStatProps & { id: string, summaryStatMessage?: string } & { children?: React.ReactNode }) {
   const { id, ...props } = propsWithID;
   return <MetricsDataProvider current id={id} >
     <SummaryMetricStatHelper {...props} />
   </MetricsDataProvider>;
 }
 
-function SummaryMetricStatHelper(props: MetricsDataComponentProps & SummaryStatProps & { children?: React.ReactNode }) {
-  const datapoints = props.data && props.data.results && props.data.results[0] && props.data.results[0].datapoints;
-  const value = datapoints && datapoints[0] && _.last(datapoints).value;
-  const {title, format} = props;
-  return <SummaryStat title={title} format={format} value={_.isNumber(value) ? value : props.value} />;
+function SummaryMetricStatHelper(props: MetricsDataComponentProps & SummaryStatProps & { summaryStatMessage?: string } & { children?: React.ReactNode}) {
+  const value = getValue(props);
+  const {title, format, summaryStatMessage} = props;
+  return (
+    <SummaryStat
+      title={title}
+      format={format}
+      value={_.isNumber(value) ? value : props.value}
+    >
+      {summaryStatMessage &&
+        <SummaryStatMessage message={summaryStatMessage}/>
+      }
+    </SummaryStat>
+  );
 }
 
+function getValue(props: MetricsDataComponentProps & SummaryStatProps) {
```

Comment:
It's hard to understand what this function does without reading the whole thing. Could you use a more descriptive name, like `aggregateLatestValuesFromMetrics` or something, and/or add a comment?